### DISCUSSION TITLE
Ditch WebView2 in Widgets panel and use native WinUI

### DISCUSSION THREAD

**Discussion by AzAgarampur** [microsoft/WindowsAppSDK#3586] [General]
### Describe the bug

The widgets board currently uses Edge's WebView2 to render content and host widget content. This should be dropped and instead a native WinUI 2/3 based interface should be used. Using WebView2 is essentially running a web browser in the background constantly and this results in (more than) a few problems:

- Widget board hangs and crashes
- These hangs slow down the entire windows explorer interface
- Very poor performance: constant laggy scrolling, blank widget content, delayed feedback from widget interaction. Dragging a widget causes the worst lag possible.
- Unnecessary CPU usage and very high memory usage
- Inconsistent theming across UI elements

Even something as simple as toggling the full screen widgets panel makes the entire UI freeze and flash multiple times while displaying no content for about 5 seconds before it slowly loads in.

---

Even this repository states that "WinUI is the native UI layer for Windows that embodies Fluent Design and delivers modern, performant, and polished user experiences..." so using WinUI for a native windows component is a no brainer.

The same could be said for other windows components as well, such as the search box experience. While it does not use WebView2, it is still a web-based experience that has the same problems listed above; WinUI would fix it too.

### Steps to reproduce the bug

1. Open the Widgets panel (Win+W)
2. Mess around with it
3. Notice the slow performance
4. Open task manager
5. Notice how WebView2 is consuming lots of memory and other system resources

### Expected behavior

Native XAML/WinUI based Widgets board that works smoothly and has a consistent theming with other windows components.

### NuGet package version

None

### Packaging type

Packaged (MSIX), Unpackaged

### Windows version

Insider Build (xxxxx), Windows 11 version 22H2 (22621, 2022 Update), Windows 11 version 21H2 (22000)

### IDE

Visual Studio 2022-preview, Visual Studio 2022, Visual Studio 2019, Other

**Comment by jaigak** [comment]
The widget panel is a joke. It's not even as helpful as widgets on iOS, despite Microsoft having live tiles 12 years ago, and only exists to push MSN content and other web based services. I strongly recommend bringing back live tiles and adding interactivity to it, rather than make something completely new.

**Comment by AzAgarampur** [comment]
Maybe live tile content can be brought out of a "tile" and be displayed as a "widget-like" experience, but something similar to tiles? Like this image:

![image](https://user-images.githubusercontent.com/42058942/232595879-2789bdee-7532-4ef7-8f2f-504960725f29.png)
(image from mspoweruser.com)

It can have all the same traditional live tile functionality but as you mentioned it can have the interactivity that widgets have now. And of course the native WinUI experience powers it.

**Comment by jaigak** [comment]
I would also prefer the tiles/widgets be always on the desktop and shown on sign-in, instead of a separate flyout.

**Comment by AzAgarampur** [comment]
So something like the UWP Notes app's window? 
![image](https://user-images.githubusercontent.com/42058942/232896991-cbb54a47-10bc-429e-b24a-19bbe7822569.png)

Honestly, I think having that Widgets side pane like Windows Vista had won't be a bad idea... so the user can either put them anywhere they want to on the desktop or can automatically arrange them in a pane.

Or, letting the user place them anywhere and having the current Windows 11 flyout can be two options the user can have.
